common/main/java/com/couchbase/lite/AbstractReplicator.java

```diff
@@ -382,9 +382,11 @@ public void removeChangeListener(@NonNull ListenerToken token) {
         }
     }
 
-    // When a replicator is closed it is detached from its from LiteCore and receives no further status updates.
-    // If it is not already in the STOPPED state, it will never be in the stopped state.  That means that
-    // a database that holds a reference to it can never close.
+    // This is a workaround: closing at the wrong time (while a push filter is running?) might even
+    // cause a crash.  The issue it addresses is that when a replicator is closed it is detached from
+    // its from LiteCore and receives no further status updates. If it is not already in the STOPPED
+    // state, it will never be in the stopped state.  That means that a database that holds a reference
+    // to it can never close.
```

Comment:
Did I review this maybe on another PR?

Basically, this comment which discussed about the workaround and LiteCore will be shown up in the API doc. Maybe move inside the method?

Reply:
Yes.  Exactly.  It now acknowledges the possibility of a crash, as you suggested.